MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -51,6 +51,18 @@ EFI_MEMORY_ATTRIBUTE_PROTOCOL  *MemoryAttributeProtocol = NULL;
 
 #define SPECIAL_REGION_PATTERN  7732426 // 7-(S) 7-(P) 3-(E) 2-(C) 4-(I) 2-(A) 6-(L)
 
+// TRUE if A and B have overlapping intervals
+#define CHECK_OVERLAP(AStart, AEnd, BStart, BEnd) \
+  ((AStart <= BStart && AEnd > BStart) || \
+  (BStart <= AStart && BEnd > AStart))
+
+// TRUE if A interval subsumes B interval
+#define CHECK_SUBSUMPTION(AStart, AEnd, BStart, BEnd) \
+  ((AStart < BStart) && (AEnd > BEnd))
+
+// TRUE if A is a subset of B

```

Comment:
Nit: Can you add a comment that this macro is comparing EFI_MEMORY_* bitwise attributes?

Reply:
I'll put bitwise subset because this is just the generic definition of a bitwise subset